api/v1alpha3/owner_labels.go

```diff
@@ -239,6 +239,33 @@ func deleteChildrenSingle(ctx context.Context, c client.Client, childObjectList
 	return deleteRetry.withObjectList(objectList), g.Wait()
 }
 
+// DeleteChildrenWithLabelsParallel deletes all the children of a parent with the resource types defined
+// in a list of ObjectList types and the labels defined in matchingLabels. Deletion will start on
+// all child types in parallel.
+func DeleteChildrenWithLabelsParallel(ctx context.Context, c client.Client, childObjectLists []ObjectList, parent metav1.Object, matchingLabels client.MatchingLabels) (DeleteStatus, error) {
+	for label, value := range MatchingOwner(parent) {
+		matchingLabels[label] = value
+	}
+
+	complete := true
+	for _, childObjectList := range childObjectLists {
+		deleteStatus, err := deleteChildrenSingle(ctx, c, childObjectList, parent, matchingLabels)
+		if err != nil {
```

Comment:
Would it make sense to keep going to try to delete the rest of the childObjectList items and return after attempting them all?

Reply:
I don't think it matters too much. The error here would be an error communicating the Delete() to the API server, and will either be resolved quickly or indicate a larger problem with the network. Returning the error immediately rather than keeping track of which errors occurred seemed like the path of least resistance.